javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -258,23 +303,31 @@ object Character {
 
   // --- End of UTF-16 surrogate pairs handling ---
 
-  def isUnicodeIdentifierStart(c: scala.Char): scala.Boolean =
-    reUnicodeIdentStart.test(c.toString)
-
-  def isUnicodeIdentifierPart(c: scala.Char): scala.Boolean =
-    isUnicodeIdentifierStart(c) || isIdentifierIgnorable(c) ||
-    reUnicodeIdentPartExcl.test(c.toString)
-
-  def isIdentifierIgnorable(c: scala.Char): scala.Boolean =
-    reIdentIgnorable.test(c.toString)
-
-  //def isMirrored(c: scala.Char): scala.Boolean
-
   def isLowerCase(c: scala.Char): scala.Boolean =
-    reIsLower.test(c.toString)
+    isLowerCase(c.toInt)
+
+  def isLowerCase(c: scala.Int): scala.Boolean = {
+    if (c < 256) {
+      '\u00AA' == c || '\u00BA' == c || getTypeTL256(c) == LOWERCASE_LETTER
+    } else {
+      ('\u02B0' <= c && c <= '\u02B8') || ('\u02C0' <= c && c <= '\u02C1') ||
+          ('\u02E0' <= c && c <= '\u02E4') || '\u0345' == c || '\u037A' == c ||
+          ('\u1D2C' <= c && c <= '\u1D6A') || '\u1D78' == c ||
+          ('\u1D9B' <= c && c <= '\u1DBF') || '\u2071' == c || '\u207F' == c ||
+          ('\u2090' <= c && c <= '\u209C') || ('\u2170' <= c && c <= '\u217F') ||
+          ('\u24D0' <= c && c <= '\u24E9') || ('\u2C7C' <= c && c <= '\u2C7D') ||
+          '\uA770' == c || ('\uA7F8' <= c && c <= '\uA7F9') ||
+          getTypeGTE256(c) == LOWERCASE_LETTER
```

Comment:
Extract the `else` branch in a separate method, so that the fast-path method is smaller.
Also, there should be no additional indentation for the second line and following wrt. the first line of the big `||`.